crates/wiggle/borrow/src/lib.rs

```diff
@@ -84,17 +94,29 @@ impl InnerBorrowChecker {
         Ok(h)
     }
 
-    fn borrow(&mut self, r: Region) -> Result<BorrowHandle, GuestError> {
+    fn immut_borrow(&mut self, r: Region) -> Result<BorrowHandle, GuestError> {
+        if !self.mut_borrows.values().all(|b| !b.overlaps(r)) {
```

Comment:
Perhaps eliding the `!` and switching to `any`?

Reply:
Done!